frontend/libs/studio-components/src/components/StudioTextField/StudioSchemaUtils.ts

```diff
@@ -0,0 +1,41 @@
+// TODO: should updated because schema is undefiend, it is defiend just in formValidationUtiles
+import Ajv from 'ajv';
+import addFormats from 'ajv-formats';
+import type { ErrorObject } from 'ajv';
+
+const ajv = new Ajv({
```

Comment:
It seems like the schema variable is undefined because we're creating a new instance of Ajv here without specifying a schema. If you look in the file where the methods were originally fetched from, you'll see a method that adds schemas using ajv.addSchema. I actually don't want to have this dependency in the Studio Components package. I suggest that we refactor this file into a class that takes the Ajv instance as an argument from outside. This way, we can keep the logic in the file, but the schema itself and the instance can come from the outside.

Reply:
ok , i will try to refactor it using class 😊